### PR TITLE
Migrated Vision Upgrades from IARRC-2017 to Snowflake

### PR DESCRIPTION
**Pull request by garethellis0**
- Removed old `ros_vision` stuff
- Added new `HSV` and `IPM` nodes
- merged code was from `IARRC 2017` - `robyn/working_merged_code`

### CODE REVIEW

File: src/vision/src/HSVFilterNode.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
+/*
+ * Created By: Robyn Castro
+ * Created On: June 17, 2017
+ * Description: Filters out everything but green.
```
Comment (ValRat) [inline src/vision/src/HSVFilterNode.cpp]: More apt description would be like:
`Filters an image in the HSV colourspace to a binary image`

File: src/vision/src/ipm_filter.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * Description: Filters out all non-green from the image
+ *              then publishes the new filtered image.
+ */
+
+#include "../include/IPMFilterNode.h"
+#include <ros/ros.h>
+
+int main(int argc, char** argv) {
+    // Setup your ROS node
+    std::string node_name = "IPMFilter";
```
Comment (RobynCastro) [inline src/vision/src/ipm_filter.cpp]: Just realized hsv_filter node_name is in snake case. For consistency sake, should IPMFilter be changed or hsv_filter?

**Comment by RobynCastro** [comment]
Looks good, all files seem to be there, and all changes seem reasonable. Just have one minor comment on node_names inside hsv_filter.cpp and ipm_filter.cpp

**Comment by ValRat** [comment]
This closes #113 as well.

File: src/vision/src/ipm_filter.cpp (hunk 2 above)
Comment (garethellis0) [reply src/vision/src/ipm_filter.cpp]: Good point. I'm not sure if we've every explicitly looked at this and decided before, but most nodes, including the sample node, seem to be in snake case. I've changed this here, as well as in `vision/CMakeLists.txt`.

**Comment by garethellis0** [comment]
@RobynCastro - ready for re-review

**Pull request merged**